blueocean-rest-impl/src/main/java/io/jenkins/blueocean/service/embedded/util/FavoriteUtil.java

```diff
@@ -107,4 +103,28 @@ public static BlueFavorite getFavorite(Item item, @Nonnull Reachable parent){
         return null;
     }
 
+    /**
+     * Resolves the default branch for a folder
+     * @param folder to check within
+     * @return default branch
+     */
+    public static Job resolveDefaultBranch(AbstractFolder folder) {
```

Comment:
It's a pity we're not using the findbugs annotations in any of this code. Nothing to do with this PR though since it's not used anywhere else either.